core/src/main/scala/geotrellis/store/query/Repository.scala

```diff
@@ -16,11 +16,24 @@
 
 package geotrellis.store.query
 
-import cats.Id
+import cats.{FlatMap, Functor, Semigroup, SemigroupK}
+import cats.syntax.functor._
+import cats.syntax.flatMap._
+import cats.syntax.semigroupk._
+import geotrellis.store.query
 
 trait RepositoryM[M[_], G[_], T] {
-  def store: M[G[T]]
+  def store: M[G[T]] = find(query.all)
   def find(query: Query): M[G[T]]
 }
 
-trait Repository[G[_], T] extends RepositoryM[Id, G, T]
+object RepositoryM {
+  implicit def semigroupRepositoryM[M[_]: FlatMap: Functor, G[_]: SemigroupK, T]
+    : Semigroup[RepositoryM[M, G, T]] =
+    (x: RepositoryM[M, G, T], y: RepositoryM[M, G, T]) =>
+      new RepositoryM[M, G, T] {
+
+        override def find(query: Query): M[G[T]] =
+          x.find(query).flatMap(g => y.find(query).map(_ <+> g))
```

Comment:
what do you think about the ordering of this combine? mb it should be `g <+> _`?

Reply:
Oooooh, probably we don't need `Functor`. `FlatMap` provides you a [map2](https://typelevel.org/cats/api/cats/FlatMap.html#map2[A,B,Z](fa:F[A],fb:F[B])(f:(A,B)=%3EZ):F[Z]) function, so we can use that in composition.

Reply:
^ I didn't try the above so i can be wrong

Reply:
That does it...and is so much cleaner. That's a nifty shortcut.